### PR TITLE
LPS-175667 Add tests on expose permissions for individual object entry

### PR DESCRIPTION
**Pull request by yanan**
**Background:**
- Add tests for LPS-162878 Expose permissions for individual object entries

**Test Plan**
- Given Object Definition student created
And Given student entry created
When requesting getStudentPermissionsPage with created {studentId}
Then I can see roles with permissions assigned to the student entry
- Given Object Definition student created
And Given student entry created
When requesting putStudentPermissionsPage with created {studentId} to update actionIds of "Owner"
Then I can see "Owner" actionIds are updated
- Given Object Definition student created
And Given student entry created
And Given new role "Test" with permissions created
And Given the role "Test" assigned to student entry
When requesting getStudentPermissionsPage with created {studentId} and roleNames as "Test"
Then I can see roles with permissions assigned to the student entry
- Given Object Definition student created
And Given student entry created
And Given new role "Test" with permissions created
And Given the role "Test" assigned to student entry
And Given with putStudentPermissionsPage I update "Test" role with no actionIds
When requesting getStudentPermissionsPage with created {studentId}
Then I can see "Owner" as a unique assigned role

**Jira Ticket:**
- https://issues.liferay.com/browse/LPS-175667

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
Failed tests are unrelated to the pr, the only failure is failed to start to run test
image

**Review by Hijky — APPROVED** [review]

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro [hunk 1]
```diff
@@ -0,0 +1,65 @@
+definition {
+
+	macro _curlObjectPermissions {
+		Variables.assertDefined(parameterList = "${en_US_plural_label},${objectEntryId}");
+
+		var en_US_plural_label_lowercase = StringUtil.lowerCase(${en_US_plural_label});
+		var portalURL = JSONCompany.getPortalURL();
+		var headerAndType = '''
```
[the remark this answers is not in the document]
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro]: good idea, I've updated and rebased to master :smile:

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Review by Hijky — APPROVED** [review]

File: modules/apps/object/object-rest-test/src/testFunctional/tests/ExposePermissionsForIndividualObjectEntries.testcase [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+				name = "Student",
+				requiredStringFieldName = "name");
+		}
+
+		task ("And Given student entry created") {
+			var studentEntryId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "students",
+				name = "Able");
+
+			CustomObjectAPI.staticObjectEntryId(objectEntryId = ${studentEntryId});
```
[the remark this answers is not in the document]
Comment (Hijky) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/ExposePermissionsForIndividualObjectEntries.testcase:25]: @magjed4289 @yanan For this, I have a question if we need to move all the macros about custom objects to CustomObjectAPI

[the remark this answers is not in the document]
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/ExposePermissionsForIndividualObjectEntries.testcase:25]: I think so, we'll move custom objects(entry) related methods into this file in the future

**Comment by yanan** [comment]
ci:test:sf

**Pull request closed without merging**